internal/cloudplugin/cloudplugin1/grpc_client.go

```diff
@@ -0,0 +1,72 @@
+package cloudplugin1
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/hashicorp/terraform/internal/cloudplugin"
+	"github.com/hashicorp/terraform/internal/cloudplugin/cloudproto1"
+)
+
+// GRPCCloudClient is the client interface for interacting with terraform-cloudplugin
+type GRPCCloudClient struct {
+	client  cloudproto1.CommandServiceClient
+	context context.Context
+}
+
+// Proof that GRPCCloudClient fulfills the go-plugin interface
+var _ cloudplugin.Cloud1 = GRPCCloudClient{}
+
+// Execute sends the client Execute request and waits for the plugin to return
+// an exit code response before returning
+func (c GRPCCloudClient) Execute(args []string, stdout, stderr io.Writer) int {
+	client, err := c.client.Execute(c.context, &cloudproto1.CommandRequest{
+		Args: args,
+	})
+
+	if err != nil {
+		fmt.Fprint(stderr, err.Error())
+		return 1
+	}
+
+	for {
+		// cloudplugin streams output as multiple CommandResponse value. Each
+		// value will either contain stdout bytes, stderr bytes, or an exit code.
+		response, err := client.Recv()
+		if err == io.EOF {
+			log.Print("[DEBUG] received EOF from cloudplugin")
+			break
+		} else if err != nil {
+			fmt.Fprintf(stderr, "Failed to receive command response from cloudplugin: %s", err)
```

Comment:
This isn't a logging behavior I'm trying to add. The plugin interface is to allow the server to repeatedly write to stdout, stderr, and then return an exit code. When the plugin server can't be contacted, the client fulfills that interface as best it can, writing to stderr and returning 1.

Reply:
we could also directly return and skip reporting `log.Print("[DEBUG] received EOF from cloudplugin")` because EOF is not technically an error here

Reply:
Also, did you notice if the other plugin client in terraform was handling the status code of this error? 🤔 I thought that on Friday I had seen somewhere in terraform that they were checking if the stream's error was unavailable to retry again.  Maybe not.